x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -219,6 +219,17 @@ export async function installComponentAndIndexTemplateForDataStream({
   componentTemplates: TemplateMap;
   indexTemplate: IndexTemplateEntry;
 }) {
+  // update index template first in case TSDS was removed, so that it does not become invalid
+  const existingIndexTemplate = await esClient.indices.getIndexTemplate({
```

Comment:
Does this API throw an error if the template doesnt exist? 

Installing the index template first isn't great but I do think it is pretty safe in the update scenario. Obviously in the install scenario we couldn't fo it in this order as the index template references the component templates which wouldn't exist.

This does mean in the future if the component templates ever changed on upgrade e.g a new one was added on package upgrade then this wouldn't work, but that is completely hypothetical so I think this is fine.

Reply:
> Does this API throw an error if the template doesnt exist?

I want to test that scenario, might have to catch a 404 error.

> This does mean in the future if the component templates ever changed on upgrade e.g a new one was added on package upgrade then this wouldn't work, but that is completely hypothetical so I think this is fine.

Good point, I think it is best to catch any errors for the first index template creation, as the codes does it again after component template update.

Reply:
Updated the code with error handling, and added integration test.